packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -49,25 +78,129 @@ export function Comments( {
-				? getBlockAttributes( _clientId )?.blockCommentId
-				: null,
-		};
-	}, [] );
+	const { toggleBlockHighlight, flashBlock } =
+		useDispatch( blockEditorStore );
+
+	const clearThreadFocus = () => {
+		// Clear any block highlights when clearing thread focus
+		if ( focusThread && blocks ) {
+			const relatedBlockClientId = findBlockByCommentId(
+				focusThread,
+				blocks
+			);
+			if ( relatedBlockClientId ) {
+				toggleBlockHighlight( relatedBlockClientId, false );
+			}
+		}
+		setFocusThread( null );
+		setShowCommentBoard( false );
+	};
 
 	const [ focusThread, setFocusThread ] = useState(
 		showCommentBoard && blockCommentId ? blockCommentId : null
 	);
 
-	const clearThreadFocus = () => {
-		setFocusThread( null );
+	// Sync comment ↔ block highlighting bidirectionally.
+	useEffect( () => {
+		const relatedBlockClientId =
+			focusThread && blocks
+				? findBlockByCommentId( focusThread, blocks )
+				: null;
+
+		// Clear highlight if switching blocks.
+		if (
+			relatedBlockClientId &&
+			relatedBlockClientId !== selectedBlockClientId
+		) {
+			toggleBlockHighlight( relatedBlockClientId, false );
+		}
+
+		// Highlight comment when block is selected.
+		if ( blockCommentId && ! focusThread ) {
+			setFocusThread( blockCommentId );
+		}
+
+		// Flash block when comment is focused (non-persistent highlight).
+		if ( relatedBlockClientId ) {
+			flashBlock( relatedBlockClientId );
+		}
+	}, [
+		selectedBlockClientId,
+		blockCommentId,
+		focusThread,
+		blocks,
+		toggleBlockHighlight,
+		flashBlock,
+		setFocusThread,
+	] );
+
+	// Clear highlights when switching to blocks without comments
+	useEffect( () => {
+		// Only clear if we're switching from a block WITH comments to a block WITHOUT comments
+		// This prevents clearing when switching between comments on the same block without comments
+		if (
+			selectedBlockClientId &&
+			! blockCommentId &&
+			focusThread &&
+			blocks
+		) {
+			// Check if the current focus thread belongs to a different block
+			const currentFocusedThreadBlock = findBlockByCommentId(
+				focusThread,
+				blocks
+			);
+			const isDifferentBlock =
+				currentFocusedThreadBlock &&
+				currentFocusedThreadBlock !== selectedBlockClientId;
+
+			if ( isDifferentBlock ) {
+				// Clear the block highlight
+				toggleBlockHighlight( currentFocusedThreadBlock, false );
+			}
+		}
+	}, [
+		selectedBlockClientId,
+		blockCommentId,
+		focusThread,
+		setFocusThread,
+		blocks,
+		toggleBlockHighlight,
+	] );
+
+	// Handle comment selection.
+	const handleCommentSelect = ( threadId ) => {
+		// Clear previous highlight if switching to a different comment.
+		if ( focusThread && focusThread !== threadId && blocks ) {
+			const relatedBlockClientId = findBlockByCommentId(
+				focusThread,
+				blocks
+			);
+			if ( relatedBlockClientId ) {
+				toggleBlockHighlight( relatedBlockClientId, false );
+			}
+		}
+
+		// Clear the add comment form when selecting a comment.
```

Comment:
```suggestion
		// Clear the add comment form when selecting a comment.
```

Now we're using `flashBlock`, so I think we don't need  `toggleBlockHighlight`.